openedx/openedx-platform
# fix: enroll users when added to a forum role

## Description

The course team management section under Instructor > Membership tab allows users to be added a role even if are not enrolled in the course. This is not the expected behaviour based on the help text displayed in the section.

This PR updates update_forum_role_membership api to enroll user before adding them to a role.

## Supporting information

`Private-ref`: [BB-7543](https://tasks.opencraft.com/browse/BB-7543)

## Testing instructions

* Setup master devstack or tutor locally.
* Register two users, one of which should have staff access.
* Using the staff user, enroll to a course.
* Under `Instructor > Membership` tab, go to `Course Team Management` section.
* Adding unenrolled user to any forum related role will enroll them to course. 

![image](https://github.com/openedx/edx-platform/assets/10894099/7e636e81-cdc0-4e15-a5d3-3d58c36a9096)


## Other information

**A point of discussion**: _Should we revoke user's course related roles when they unroll from a course?_
